### ISSUE TITLE
Draw interferes dragPan.disable()

### ISSUE THREAD

**Issue by rvratner** [mapbox/mapbox-gl-draw#654]
**mapbox-gl-js version**: 0.37
**mapbox-gl-draw version**: v0.17.4

### Steps to Trigger Behavior

 1. Load map.
 2. Load Draw control
 3. map.drapPan.disable()

### Expected Behavior
User can't drag pan the map

### Actual Behavior
Map is able to be dragpanned

If you comment out the 
            var draw = new MapboxDraw({
                displayControlsDefault: false,
                controls: {
                    polygon: true,
                    trash: true
                }
            });
You are able to disable dragPan as expected

**Comment by Elefantz** [comment]
Hi,

Same problem here. Did you find a workaround ?

Thanks,

**Comment by rvratner** [comment]
No, I never did - was hoping to hear from the folks who developed this to see if they had any insight?

**Comment by mcwhittemore** [comment]
Ah! I keep hoping to have time to really dive into this. Sorry.

**Comment by mcwhittemore** [comment]
OK. I took a quick look. This bug is happening in `simple_select` and `direct_select` but not that drawing modes. This makes me guess that we're enabling dragPan as some part of these two modes. I don't have much time right now to dive in deeper, but maybe that bit of info can help others push on this a bit.

**Comment by rvratner** [comment]
Thanks, appreciate you looking into it. Let me know if there's anything I can do on my end?

**Comment by mcwhittemore** [comment]
@rvratner - If you have time to debug this a bit and make a PR when you find a solution that would be great.

Here are the three places Draw calls `dragPan.disable()` in these two modes. 

- https://github.com/mapbox/mapbox-gl-draw/blob/master/src/modes/simple_select.js#L215
- https://github.com/mapbox/mapbox-gl-draw/blob/master/src/modes/simple_select.js#L155
- https://github.com/mapbox/mapbox-gl-draw/blob/master/src/modes/direct_select.js#L32

I'd work on this by [starting the debug page](https://github.com/mapbox/mapbox-gl-draw#developing-and-testing) and adding some logging to the modes.

**Comment by GadyTal** [comment]
Any solution?

**Comment by GadyTal** [comment]
In my case, this issue affects the draggable markers on the map, which they can't be dragged - because of the **dragPan** option can't be disabled as mapbox gl js required in this case.

**### Workaround** 

In every time that **'mouseenter'** event is fired and we should prepare the markers for drag - change draw mode by to draw_polygon (which doesn't cause the error) - 
`placeDraw.changeMode('draw_polygon');`

then, when **mouseleave** fired - change draw mode to simple_select.
`placeDraw.changeMode('simple_select');`

**Comment by amirhossein1414** [comment]
**MyWorkaround**
I solved this problem by modifying **mapbox-gl-draw.js** file.
I searched  for "dragPan.enable()" statement and delete it from mapbox-gl-draw.js  file.
If you dont want to delete it entirely, you can wrap "dragPan.enable()" in an if clause so you can
check the mapboxgl interaction option and then call dragPan.enable().

**Comment by john-melin** [comment]
This is still an issue. Is this looked being looked at?

**Comment by wandongming** [comment]
any update?

**Comment by Aiamjay** [comment]
any update

**Comment by IlariaCallegari** [comment]
Hi there, is there any update on this? Mapbox-draw is still interfering with dragPan. Any chance you'll merge PR #1191 ?

**Comment by tredpath** [comment]
This problem affects more than just `dragPan`. I tried the work around suggested [here](https://github.com/mapbox/mapbox-gl-draw/issues/654#issuecomment-423138762) which does fix `dragPan` but `doubleClickZoom` is still a problem. If you double click on the map it won't zoom immediately but it will re-enable both `dragPan` and `doubleClickZoom`.

In order to disable interactions I had to remove the draw control from the map when I wanted to disable interactions and add it back in when I wanted to re-enable them.
`map.removeControl(drawControl)`.
